sandy/sections/mf2.py

```diff
@@ -0,0 +1,367 @@
+                        "List" : np.array(L.B).reshape(NRS,6),
+                        }
+                out.update(add)
+                print (NRS)
+            
+            else :                
+                T, i = sandy.read_tab1(df, i)
+                add = {
+                        "NR": T.NBT,    
+                        "NP": T.INT,
+                        "E_int": T.x,
+                        "AP": T.y,
+                        }
+                out.update(add)
+        
+        if LRF == 3 :
+            
+            if NRO == 0 :
+               C, i = sandy.read_cont(df, i)
+               add = {
+                       "SPI": C.C1,
+                       "AP": C.C2,
+                       "LAD": C.L1,     #Flag indicating whether these parameters can be used to compute angular distributions
+                       "NLS": C.N1,     
+                       "NLSC": C.N2,    #Number of l-values
+                       }
+               NLSC = int(C.N2)
+               out.update(add)
+               for j in range(NLSC) :
+                   L, i = sandy.read_list(df, i)
+                   NRS= int(L.N2)
+                   add = {
+                            "AWRI": L.C1,
+                            "QX": L.C2,
+                            "L": L.L1,
+                            "LRX": L.L2,
+                            "6*NRS": L.NPL,
+                            "NRS": L.N2,
+                            "List": L.B,
+                            }
+                   print (NRS)
+                   out.update(add)
+               """ the code here isn't complete"""
+               
+            else :
+                T, i = sandy.read_tab1(df, i)
+                add = {
+                        "NR": T.NBT,    
+                        "NP": T.INT,
+                        "E_int": T.x,
+                        "AP": T.y,
+                        }
+                out.update(add)
+        
+        if LRF == 4 :
+            
+            
+            T, i = sandy.read_tab1(df, i)
+            add = {
+                    "NR": T.NBT,    
+                    "NP": T.INT,
+                    "E_int": T.x,
+                    "AP": T.y,
+                    }
+            out.update(add)
+            C, i = sandy.read_cont(df, i)
+            add = {
+                    "SPI": C.C1,
+                    "AP": C.C2,
+                    "NLS": C.N1,
+                    }
+            out.update(add)
+            L, i = sandy.read_list(df, i)
+            NPP= int(L.N2)
+            add = { 
+                    "AWRI": L.C1,       #Number of sets of background constants given.
+                    "LI": L.L1,         #Flag to indicate the kind of parameters given
+                    "6*NX": L.NPL,
+                    "NX": L.N2,         #Number of sets of background constants given.
+                    "AT1": L.B[0],      #Background constants for the total cross section
```

Comment:
Maybe is better "AT": L.B[0:4].

Reply:
this part of the code is now gone because LRF = 4 is outdated